docs/modules/ROOT/pages/async-context.adoc

```diff
@@ -6,71 +6,103 @@ ifndef::env-github[]
 
 Although asynchronous function composition is not required to use ServiceTalk, the internals of ServiceTalk uses
 asynchronous function composition as a core abstraction which can be difficult to reason about relative to
 traditional imperative programming. On top of this, ServiceTalk provides isolation from the EventLoop
 threads by offloading to application level threads, which introduces asynchronous boundaries. For request/response
-protocols the goal is for the `AsyncContext` to be isolated for each request/response, but folks that need to directly
-interact with `AsyncContext` need to understand how modifications are visible (or not) relative to where they are set.
+protocols the goal is for the asynchronous context to be isolated for each request/response, but folks that need to
+directly interact with asynchronous context need to understand how modifications are visible (or not) relative to where
+they are set or read in their application.
+
+== `AsyncContext`
+
+Although `AsyncContext` may be convenient to use it shouldn't be overused when traditional argument passing is an
+option. The intended use case of `AsyncContext` is to propagate context across API boundaries which do not accommodate
+for additional parameters.
+
+`AsyncContext` is designed to provide a static API to retain state associated across asynchronous boundaries that can
+be used instead of creating a traditional `ThreadLocal`.
 
 **Cost of retention**
 
 In order to make state available in a static fashion across API calls the mechanism to do this is to use thread local
-state. The same thread maybe used to process logically independent asynchronous control flows (e.g. EventLoop can
+state. The same thread maybe used to process logically independent asynchronous control flows (e.g. `EventLoop` can
 process multiple sockets, and each socket can process multiple requests, and offloading may re-use threads to process
 different requests). This results in frequently accessing the thread local state to save original value, set correct
 context, and then restore original value.
 
-== Usage
+=== Usage
 
 ServiceTalk provides out-of-the-box support for `AsyncContext`, details about which can be found
 xref:{page-version}@servicetalk-concurrent-api::async-context.adoc[here]. Users can
 xref:{page-version}@servicetalk-concurrent-api::async-context.adoc#disable-asynccontext[disable `AsyncContext`] if they
 do not require this functionality.
+
+== `CapturedContext`
+
+In contrast to the `AsyncContext`, the `CapturedContext` APIs are most suitable for capturing external context and
+restoring it within the ServiceTalk execution chain. It is _not_ a good tool for defining new contextual information.
+
+**Interop with existing APIs depending on `ThreadLocal`**
+
+Some APIs / features assume static state that is coupled with the current thread. This worked well in the
+1-request-per-thread model, however it breaks down when we may process a request on different threads due to
```

Comment:
IIRC it's typically named "1-thread-per-request model"

Reply:
I think you're right, other than the 1 is usually implied.